rpcmetrics/metrics.go

```diff
@@ -0,0 +1,103 @@
+package rpcmetrics
+
+import (
+	"sync"
+
+	"github.com/uber/jaeger-lib/metrics"
+)
+
+const defaultMaxNumberOfEndpoints = 200
+
+// Metrics is a collection of metrics for an endpoint describing
+// throughput, success, errors, and performance
+type Metrics struct {
+	// Requests is a counter of the total number of successes + failures.
+	Requests metrics.Counter `metric:"requests"`
+
+	// Success is a counter of the total number of successes.
+	Success metrics.Counter `metric:"success"`
```

Comment:
I used the naming from Finagle metrics, and they had it as Success. The naming is to be discussed anyway, e.g. prometheus docs call for much more descriptive names like `request_count_total`